src/analytics/src/main/scala/osmesa/analytics/oneoffs/MergeChangesets.scala

```diff
@@ -0,0 +1,163 @@
+
+      val endTimeOpt =
+        Opts
+          .option[Instant]("end-time",
+                       short = "e",
+                       metavar = "timestamp",
+                       help = "Timestamp of stream end (of the form 2016-02-29T13:45:00Z); if absent, the time now will be used")
+                         .orNone
+
+      val orcArg = Opts
+        .argument[URI]("source ORC")
+        .validate("URI to ORC must have an s3 or file scheme") { _.getScheme != null }
+        .validate("orc must be an S3 or file Uri") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      val outputArg = Opts.argument[URI]("destination ORC")
+        .validate("Output URI must have a scheme") { _.getScheme != null }
+        .validate("Output URI must have an S3 or file scheme") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      (changesetSourceOpt,
+       endTimeOpt,
+       orcArg,
+       outputArg).mapN {
+        (changesetSource, endTime, orcUri, outputURI) =>
+        implicit val spark: SparkSession = Analytics.sparkSession("MergeChangesets")
+
+        import spark.implicits._
+
+        val df = spark.read.orc(orcUri.toString)
+        val lastModified = df.select(max(coalesce('closed_at, 'created_at))).first.getAs[Timestamp](0)
+
+        val startSequence = findSequenceFor(lastModified.toInstant, changesetSource)
+        val endSequence = endTime.map(findSequenceFor(_, changesetSource)).getOrElse(getCurrentSequence(changesetSource).get.sequence)
+
+        val options = Map(
+          Source.BaseURI -> changesetSource.toString,
+          Source.StartSequence -> startSequence.toString,
+          Source.EndSequence -> (endSequence + 1).toString // sequence range is (]; end sequence is exclusive
+        )
+
+        val changesets = spark.read.format(Source.Changesets).options(options).load
+
+        // TODO: Clean up the following by providing and using a function in VP to coerce the
+        // column names into camel case (see https://github.com/geotrellis/vectorpipe/issues/113)
+        changesets
+          .drop("comments", "sequence")
+          .union(df.select(
+            'id,
+            'tags,
+            'created_at as 'createdAt,
+            'open,
+            'closed_at as 'closedAt,
+            'comments_count as 'commentsCount,
+            'min_lat as 'minLat,
+            'max_lat as 'maxLat,
+            'min_lon as 'minLon,
+            'max_lon as 'maxLon,
+            'num_changes as 'numChanges,
+            'uid,
+            'user)
+          )
+          .repartition(1)
+          .write
+          .orc(outputURI.toString)
+
+        spark.stop()
+      }
+    }
+)
+
+object MergeChangesetUtils {
```

Comment:
Since this is used elsewhere, should it be outside of the `oneoffs` package?

Reply:
There is so much reorganization of this repo that needs to happen.  I can see breaking out oneoffs into streaming, batch, and utilities.